framework/src/base/Assembly.C

```diff
@@ -2769,12 +2734,15 @@ Assembly::addResidual(const std::map<TagName, TagID> & tags)
 }
 
 void
-Assembly::addResidualNeighbor(NumericVector<Number> & residual, TagID tag_id /* = 0 */)
+Assembly::addResidualNeighbor(NumericVector<Number> & residual, TagID tag_id)
 {
   const std::vector<MooseVariableFEBase *> & vars = _sys.getVariables(_tid);
   for (const auto & var : vars)
-    addResidualBlock(
-        residual, _sub_Rn[tag_id][var->number()], var->dofIndicesNeighbor(), var->scalingFactor());
+    addResidualBlock(residual,
+                     _sub_Rn[tag_id][var->number()],
+                     var->dofIndicesNeighbor(),
+                     var->arrayScalingFactor(),
```

Comment:
The old method `scalingFactor` is kept. The difference is that this function returns a vector which in most cases in in size of 1. I cannot use the same name here because the only interface difference is the returned value. Better suggestions?

Reply:
I think it's fine